src/solvers/flattening/boolbv_array.cpp

```diff
@@ -6,39 +6,30 @@ Author: Daniel Kroening, kroening@kroening.com
 
 \*******************************************************************/
 
-
 #include "boolbv.h"
 
 bvt boolbvt::convert_array(const exprt &expr)
 {
-  std::size_t width=boolbv_width(expr.type());
-
+  const std::size_t width = boolbv_width(expr.type());
   if(width==0)
     return conversion_failed(expr);
 
   if(expr.type().id()==ID_array)
   {
-    assert(expr.has_operands());
+    INVARIANT(expr.has_operands(), "array should have operands");
     const exprt::operandst &operands=expr.operands();
-    assert(!operands.empty());
-    std::size_t op_width=width/operands.size();
-
+    const std::size_t op_width = width / operands.size();
     bvt bv;
     bv.reserve(width);
-
-    forall_expr(it, operands)
+    for(const exprt &op : operands)
```

Comment:
Where there is a macro; please use it.  If we are going to change whether we use them it should be one decision not people doing their own thing.

Reply:
I don't see any advantage in using the macro here since we are not using the iterator as such. I'm not saying we should never use them but in that particular case it seems wrong.